contracts/src/protocol/OmniPortal.sol

```diff
@@ -233,33 +258,43 @@ contract OmniPortal is IOmniPortal, IOmniPortalAdmin, OwnableUpgradeable, OmniPo
         emit FeeOracleChanged(oldFeeOracle, feeOracle);
     }
 
+    /**
+     * @notice Add a new validator set.
+     * @dev Only callable via xcall from Omni's consensus chain
+     * @param valSetId      Validator set id
+     * @param validators    Validator set
+     */
+    function addValidatorSet(uint64 valSetId, XTypes.Validator[] memory validators) external {
```

Comment:
can this be an internal function? Or does using address.call prevent that

Reply:
it prevents from using address(this).call
also it's part of the abi now, which helps with packing off chain (not that this would be a reason to do it)